Dependent library updates and NIO.2 for performance

adamretter opened the pull request:
In the interests of improving performance:

* Updated all Maven plugins to the latest versions
* Fixed problems with dependencies in the pom.xml files
* Updated all dependent libraries to the latest version
* Introduced OWASP into the Maven build to spot security issues with dependent libraries
* Switched DBCP for HikariCP
* Switched from OpenCSV to uniVocity
* Switched all java code (except droid-gui) over to use NIO.2
* All I/O is now buffered to improve performance

rhubner commented:
Wow, great pull req. I will look on it. I'm not sure if we can include this to 6.4.

adamretter commented:
@rhubner Okay so I fixed the build. I believe that all tests pass now. I also updated the Travis settings, the builds should be faster and are now tested and pass on JDK 1.7 and JDK 1.8.

rhubner commented:
![](https://github.trello.services/images/mini-trello-icon.png) [RH to write to AR requesting that he split the pull request into smaller chunks and fixes the error in the current version](https://trello.com/c/Ee34SbfR/98-rh-to-write-to-ar-requesting-that-he-split-the-pull-request-into-smaller-chunks-and-fixes-the-error-in-the-current-version)
